mypy/subtypes.py

```diff
@@ -293,7 +295,72 @@ def visit_instance(self, left: Instance) -> bool:
                 # Map left type to corresponding right instances.
                 t = map_instance_to_supertype(left, right.type)
                 nominal = True
-                for lefta, righta, tvar in zip(t.args, right.args, right.type.defn.type_vars):
+                if right.type.has_type_var_tuple_type:
+                    left_prefix, left_middle, left_suffix = (
+                        split_with_instance(left)
+                    )
+                    right_prefix, right_middle, right_suffix = (
+                        split_with_instance(right)
+                    )
+
+                    left_unpacked = extract_unpack(
+                        left_middle
+                    )
+                    right_unpacked = extract_unpack(
+                        right_middle
+                    )
+
+                    # Helper for case 2 below so we can treat them the same.
+                    def check_mixed(
+                        unpacked_type: ProperType,
+                        compare_to: Tuple[Type, ...]
+                    ) -> bool:
+                        if isinstance(unpacked_type, TypeVarTupleType):
+                            return False
+                        if isinstance(unpacked_type, AnyType):
+                            return True
+                        assert False
+
+                    # Case 1: Both are unpacks, in this case we check what is being
+                    # unpacked is the same.
+                    if left_unpacked is not None and right_unpacked is not None:
+                        if not is_equivalent(left_unpacked, right_unpacked):
+                            return False
+
+                    # Case 2: Only one of the types is an unpack.
+                    elif left_unpacked is not None and right_unpacked is None:
+                        if not check_mixed(left_unpacked, right_middle):
+                            return False
+                    elif left_unpacked is None and right_unpacked is not None:
+                        if not check_mixed(right_unpacked, left_middle):
+                            return False
+
+                    # Case 3: Neither type is an unpack. In this case we just compare
+                    # the items themselves.
+                    else:
+                        if len(left_middle) != len(right_middle):
+                            return False
+                        for left_t, right_t in zip(left_middle, right_middle):
+                            if not is_equivalent(left_t, right_t):
+                                return False
+
+                    left_items = t.args[:right.type.type_var_tuple_prefix]
+                    right_items = right.args[:right.type.type_var_tuple_prefix]
+                    if right.type.type_var_tuple_suffix:
+                        left_items += t.args[-right.type.type_var_tuple_suffix:]
+                        right_items += right.args[-right.type.type_var_tuple_suffix:]
```

Comment:
@jhance maybe I'm just totally missing something, but I think these aren't actually used anywhere? The slicing means we're not mutating the original `t.args` and `right.args`